### ISSUE TITLE
Some problem detected during publishing

### ISSUE THREAD

**Issue by kirill-grouchnikov** [vanniktech/gradle-maven-publish-plugin#237]
```
> Task :icon:transcoder:publishMetadataPublicationToMavenCentralRepository
Execution optimizations have been disabled for task ':icon:transcoder:publishMetadataPublicationToMavenCentralRepository' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\Projects\aurora\icon\transcoder\build\libs\aurora-transcoder-metadata-0.0.29-SNAPSHOT.jar.asc'. Reason: Task ':icon:transcoder:publishMetadataPublicationToMavenCentralRepository' uses this output of task ':icon:transcoder:signKotlinMultiplatformPublication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\Projects\aurora\icon\transcoder\build\libs\transcoder-0.0.29-SNAPSHOT-javadoc.jar.asc'. Reason: Task ':icon:transcoder:publishMetadataPublicationToMavenCentralRepository' uses this output of task ':icon:transcoder:signDesktopPublication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\Projects\aurora\icon\transcoder\build\libs\transcoder-0.0.29-SNAPSHOT-javadoc.jar.asc'. Reason: Task ':icon:transcoder:publishMetadataPublicationToMavenCentralRepository' uses this output of task ':icon:transcoder:signKotlinMultiplatformPublication' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#implicit_dependency for more details about this problem.
```

This is running `gradlew publish` with 0.15.0

**Comment by kirill-grouchnikov** [comment]
This is shown on multiple modules in https://github.com/kirill-grouchnikov/aurora

Going back to 0.14.2 doesn't show these messages, so something changed in 0.15.0

**Comment by kirill-grouchnikov** [comment]
Not sure what .asc files are, but definitely 0.15.0 is creating a bunch of them in https://oss.sonatype.org/content/repositories/snapshots/org/pushing-pixels/aurora-component-desktop/0.0.29-SNAPSHOT/ whereas 0.14.2 does not in https://oss.sonatype.org/content/repositories/snapshots/org/pushing-pixels/aurora-component-desktop/0.0.30-SNAPSHOT/

Note that the 0.0.29 folder has two sets of files from two attempts to run `gradlew publish`

**Comment by gabrielittner** [comment]
The `.asc` files are the output of the signing task. The reason why you get them in 0.15.0 and didn't before for snapshots is that in the past we explicitly checked to not sign snapshots. There isn't really a point in doing that though, it doesn't hurt and you can also test your signing set up using snapshots this way. The only thing that is special about snapshots now is that signing wouldn't fail if it's not configured while it does for release versions. 

This is also why you are seeing the warnings above in 0.15.0 but not with 0.14.2, if you'd run `publishToMavenLocal` with a non snapshot version you also get those warnings on 0.14.2. I will test if the warnings come from something on our side or on Gradle's side. While we do tell Gradle what to sign, they do the actual task wiring so I assume it's on their side.

**Comment by kirill-grouchnikov** [comment]
I don't mind the warnings in the terminal, but the end result of it was that it looked like some of the artifacts were not fully uploaded to Sonatype staging, and I couldn't use them in a demo project.

**Comment by gabrielittner** [comment]
I'll also look into that 👍 Have you by chance tried multiple times to check if it happens everytime that some are missing? Just to rule out flakiness on the Sonatype side

**Comment by kirill-grouchnikov** [comment]
I might have tried the demo project right after uploading the binaries to staging. Tried today with 0.0.29 created with version 0.15.0 of this plugin, and with 0.0.30 created with version 0.14.2 of this plugin. Both work. 

So is it safe to ignore these terminal blobs?

**Comment by gabrielittner** [comment]
I was able to reproduce the warnings when using the plain `maven-publish` plugin, so I filed an issue here https://github.com/gradle/gradle/issues/17043

I would say it's safe to ignore the warnings. It's been there before, the only thing that changed is that Gradle 7 started surfacing the issue and 0.15.0 makes it more prominent by also signing snapshots by default

**Comment by gabrielittner** [comment]
The warnings seem to have been resolved in Kotlin 1.5.0: https://youtrack.jetbrains.com/issue/KT-46466
